tests/tests/tier0/bluechi-reload-unit-service/test_bluechi_reload_unit_service.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import logging
+import time
+
+from typing import Dict
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+from bluechi_test.util import read_file
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+failed_status = "failed"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+    service_file_location = "/usr/lib/systemd/system/"
+    new_service_str = "service/new-service.service"
+
+    c_code_location = "/var/tmp/"
+    invalid_conf_str = "code/new-service-code.c"
+
+    content = read_file(new_service_str)
+    node_foo.create_file(service_file_location, new_service_str, content)
+
+
+    print(node_foo.exec_run("systemctl start new-service.service")[1])
+    output = ctrl.exec_run(f"bluechictl list-units {NODE_FOO} ")[1].split("\n")
+
+    for line in output:
+        if "new-ser" in line:
+            print(line)
+
+    node_foo.exec_run(f"sed -i '/ExecStart=/c\\ExecStart=/bin/true' "
+                              f"/usr/lib/systemd/system/new-service.service")
+    print(node_foo.exec_run("systemctl daemon-reload ")[1])
+    print(node_foo.exec_run("systemctl start new-service.service")[1])
+    print(ctrl.exec_run(f"bluechictl reload {NODE_FOO} new-service.service")[1])
+
+    output = ctrl.exec_run(f"bluechictl list-units {NODE_FOO} ")[1].split("\n")
+
+    for line in output:
+        if "new-ser" in line:
+            print(line)
```

Comment:
```suggestion
    # create the service file
    content = read_file(os.join.path("systemd", "new-service.service"))
    self.create_file(os.path.join("/", "etc", "systemd", "system"), "new-service.service", content)
    
    # start the service via bluechictl, it should fail since the service hasn't been found
    result, output = ctrl.bluechictl.start_unit(NODE_FOO, "new-service.service", check_result=False)
    assert result == 1
    assert "not found" in output
    
    # reload the systemd daemon
    ctrl.bluechictl.reload_on_node(NODE_FOO)
    
    # start the service via bluechictl, it should fail since the service hasn't been found
    result, output = ctrl.bluechictl.start_unit(NODE_FOO, "new-service.service", check_result=False)
    assert result == 0
    assert "not found" not in output
```

This is roughly the flow for a systemd daemon reload. 
It requres to the `reload_on_node` function to be added to `BluechiCtl` like this:
```python
# in tests/bluechi_test/bluechictl.py
# ...

def reload_on_node(self, node_name: str, check_result: bool = True, expected_result: int = 0) \
        -> Tuple[Optional[int], Union[Iterator[bytes], Any, Tuple[bytes, bytes]]]:
        return self._run(
            f"Reloading systemd daemon on node {node_name}",
            f"reload {node_name}",
            check_result,
            expected_result
        )
```